### PR TITLE
[HttpKernel] Add support for custom HTTP status code for the `#[MapQueryParameter]` attribute

### PR DESCRIPTION
**Pull request by ovidiuenache**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | #52782
| License       | MIT

Issue #52782 was created where the need to have a custom HTTP status code for the MapQueryParameter attribute was stated. As part of the discussion, @nicolas-grekas suggested an approach similar to the implementation of the MapQueryString attribute where such a status code can be provided in the attribute's constructor.

I believe having the possibility to control the status code returned for an invalid and non-nullable parameter with no default value would be a great addition so I took the liberty to create this PR.

This change is backward compatible.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: It'd be great to preserve the original http exception type.
I've submitted #53209 to help doing so.

Comment (ovidiuenache) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: I think you meant #53212 :)

So should I wait for it to get to merged and base this one on top of 7.1 and consider it a new feature?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: it's a new feature yes, please rebase

Comment (ovidiuenache) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: Rebased. I'll also update the code to preserve the original HTTP exception type once #53212 gets merged.

File: src/Symfony/Component/HttpKernel/CHANGELOG.md
Comment (OskarStark) [inline src/Symfony/Component/HttpKernel/CHANGELOG.md]:
```suggestion
 * Add `$validationFailedStatusCode` argument to `#[MapQueryParameter]` that allows setting a custom HTTP status code when validation fails
```

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php
Comment (nicolas-grekas) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: Rebase unlocked :)

Comment (ovidiuenache) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php]: Rebased. I believe we should also fix the labels for the PR.

File: src/Symfony/Component/HttpKernel/Attribute/MapQueryParameter.php
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Attribute/MapQueryParameter.php]: I'd remove readonly for consistency with the other properties. It wouldn't provide any benefits anyway in practice.

**Review by nicolas-grekas — APPROVED** [review]
LGTM with one minor comment.

**Review by OskarStark — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @ovidiuenache.

**Pull request merged**
